src/lucky_migrator/create_table_statement.cr

```diff
@@ -58,25 +58,26 @@ class LuckyMigrator::CreateTableStatement
 
   # Generates raw sql from a type declaration and options passed in as named
   # variables.
-  macro add(type_declaration, index = false, using = :btree, unique = false)
+  macro add(type_declaration, index = false, using = :btree, unique = false, default = nil)
     {% if type_declaration.type.is_a?(Union) %}
-      add_column :{{ type_declaration.var }}, {{ type_declaration.type.types.first }}, optional: true
+      add_column :{{ type_declaration.var }}, {{ type_declaration.type.types.first }}, optional: true, default: {{ default }}
     {% else %}
-      add_column :{{ type_declaration.var }}, {{ type_declaration.type }}
+      add_column :{{ type_declaration.var }}, {{ type_declaration.type }}, default: {{ default }}
     {% end %}
 
     {% if index || unique %}
       add_index :{{ type_declaration.var }}, using: {{ using }}, unique: {{ unique }}
     {% end %}
   end
 
-  def add_column(name, type : (String | Time | Int32 | Int64 | Float | Bool).class, optional = false, reference = nil, on_delete = :do_nothing)
+  def add_column(name, type : (String | Time | Int32 | Int64 | Float | Bool).class, optional = false, reference = nil, on_delete = :do_nothing, default : String | Time | Int32 | Int64 | Float | Bool | Symbol | Nil = nil)
```

Comment:
@paulcsmith  Is there anyway to refactor these long Union types? For example in Typescript you can create a variable like `ManyTypes = number | string | Object;` and use it like `ambiguousObject: ManyTypes`.

Reply:
Good questions and I think you're right about needing to extract the type. I believe you can use `alias`: https://crystal-lang.org/docs/syntax_and_semantics/alias.html

Something like: `alias ColumnType = (String | Time | Etc).class`

Reply:
Exactly what I as looking for, thanks!

Reply:
For some reason my comment I keep trying to leave here keeps getting lost haha. Do you want to try using `alias` on this?

Reply:
Good idea, I'll do that right now

Reply:
I keep getting this error on using Floats:
"can't use Float in unions yet, use a more specific type
`alias ColumnType = (String | Time | Int32 | Int64 | Float | Bool).class`"

Reply:
Ok I tried some stuff out and it seems like you need to add `.class` to each type:

```crystal
alias ColumnType = (String.class | Time.class | etc.)
```

I'm not sure why it worked before but not for alias. Hopefully this works

Reply:
It looks like I have to use Float32 or Float64 explicitly, but I'll check it out tomorrow and let you know

Reply:
What I have right now is:
```crystal
alias ColumnType = String | Time | Int32 | Int64 | Float.class | Bool

def add_column(name, type : ColumnType.class, optional = false, reference = nil, on_delete = :do_nothing, default : ColumnType.class | Nil = nil)
  ...
end
```
And this is working for all types except for Float:
undefined method add_column for `add_column :amount_paid, Float, default: nil`

Reply:
I think Float defaults to Float64, so what if we just enforce using Float64?

Reply:
Sorry I don't have tonight, but I'll work on it tomorrow and let you know what I find